client/src/globalState.tsx

```diff
@@ -0,0 +1,274 @@
+import { ApolloError, useLazyQuery, useSubscription } from "@apollo/client";
+import React, { useContext, useEffect, useState } from "react";
+import { Item, User, WorkflowStub } from "../../shared/graphql/graphql";
+import { gqlGetPinboardByComposerId, gqlOnCreateItem } from "../gql";
+import { EXPAND_PINBOARD_QUERY_PARAM } from "./app";
+import { Floaty } from "./floaty";
+import { Panel } from "./panel";
+import type { PinboardData } from "./pinboard";
+import type { PayloadAndType } from "./types/PayloadAndType";
+import type { PerPinboard } from "./types/PerPinboard";
+
+interface GlobalStateContextShape {
+  userEmail: string;
+  userLookup: { [email: string]: User } | undefined;
+
+  activePinboardIds: string[];
+  payloadToBeSent: PayloadAndType | null;
+  clearPayloadToBeSent: () => void;
+
+  openPinboard: (pinboardData: PinboardData) => void;
+  closePinboard: (pinboardId: string) => void;
+  preselectedPinboard: WorkflowStub | undefined;
+  selectedPinboardId: string | null | undefined;
+  clearSelectedPinboard: () => void;
+
+  showNotification: (item: Item) => void;
+  hasWebPushSubscription: boolean | null | undefined;
+
+  errors: PerPinboard<ApolloError>;
+  setError: (pinboardId: string, error: ApolloError | undefined) => void;
+  hasErrorOnOtherPinboard: (pinboardId: string) => boolean;
+
+  isExpanded: boolean;
+  setIsExpanded: (_: boolean) => void;
+
+  unreadFlags: PerPinboard<boolean>;
+  setUnreadFlag: (
+    pinboardId: string
+  ) => (unreadFlag: boolean | undefined) => void;
+  hasUnreadOnOtherPinboard: (pinboardId: string) => boolean;
+}
+const GlobalStateContext = React.createContext<GlobalStateContextShape | null>(
+  null
+);
+
+const GlobalStateContextProvider = GlobalStateContext.Provider;
+
+// Ugly but allows us to assume that the context has been set, which it always will be
+export const useGlobalStateContext = (): GlobalStateContextShape => {
+  const ctx = useContext(GlobalStateContext);
+  if (ctx === null) {
+    throw new Error("GlobalStateContext is uninitialised");
+  }
+  return ctx;
+};
+
+interface GlobalStateProps {
+  userEmail: string;
+  preselectedComposerId: string | null | undefined;
+  presetUnreadNotifications: number | undefined;
+  payloadToBeSent: PayloadAndType | null;
+  clearPayloadToBeSent: () => void;
+  isExpanded: boolean;
+  setIsExpanded: (_: boolean) => void;
+  userLookup: { [email: string]: User } | undefined;
+  hasWebPushSubscription: boolean | null | undefined;
+  showNotification: (item: Item) => void;
+  clearDesktopNotificationsForPinboardId: (pinboardId: string) => void;
+}
+export const GlobalState: React.FC<GlobalStateProps> = ({
```

Comment:
this name is confusing at the call site in `app.tsx`, what do you think of...
```suggestion
export const FloatyAndPanelWithGlobalState: React.FC<GlobalStateProps> = ({
```